src/Extensions/ExtHostClient.rei

```diff
@@ -4,6 +4,23 @@ module Core = Oni_Core;
 
 type t;
 
+type msg =
+  | RegisterSourceControl({
+      handle: int,
+      id: string,
+      label: string,
+      rootUri: option(Core.Uri.t),
+    })
+  | UnregisterSourceControl({handle: int})
+  | UpdateSourceControl({
+      handle: int,
+      hasQuickDiffProvider: option(bool),
+      count: option(int),
+      commitTemplate: option(string),
+    });
+// acceptInputCommand: option(_),
+// statusBarCommands: option(_),
```

Comment:
Instead of having a bunch of optional callbacks passed into `onMessage` I've added a mandatory `dispatch` function that will handle these messages. This gives us, most importantly, exhaustiveness checking, but also an opportunity for easy logging if needed, and requires less boilerplate.

Reply:
Good idea!